File: src/main/java/waterpower/client/gui/ContainerRotor.java
/**
 * Copyright (c) Huang Yuhui, 2014
 * 
 * "WaterCraft" is distributed under the terms of the Minecraft Mod Public
 * License 1.0, or MMPL. Please check the contents of the license located in
 * http://www.mod-buildcraft.com/MMPL-1.0.txt
 */

package waterpower.client.gui;

import net.minecraft.entity.player.EntityPlayer;
import waterpower.common.block.inventory.SlotInventorySlot;
import waterpower.common.block.tileentity.TileEntityRotor;

public class ContainerRotor extends ContainerFullInventory {
    public TileEntityRotor tileEntity;

    public ContainerRotor(EntityPlayer player, TileEntityRotor tileEntityCW) {
        super(player, tileEntityCW, 166);
        tileEntity = tileEntityCW;
        layoutContainer();
    }

    protected void layoutContainer() {

        // Rotor inventory drawing
        addSlotToContainer(new SlotInventorySlot(tileEntity.slotRotor, 0, 80, 26));

    }
}

File: src/main/kotlin/waterpower/common/block/container/ContainerRotor.kt
/**
 * Copyright (c) Huang Yuhui, 2017
 *
 * "WaterPower" is distributed under the terms of the Minecraft Mod Public
 * License 1.0, or MMPL. Please check the contents of the license located in
 * http://www.mod-buildcraft.com/MMPL-1.0.txt
 */
package waterpower.common.block.container

import net.minecraft.entity.player.EntityPlayer
import waterpower.common.block.inventory.SlotInventorySlot
import waterpower.common.block.tile.TileEntityRotorGenerator

open class ContainerRotor(player: EntityPlayer, val te: TileEntityRotorGenerator)
    : ContainerFullInventory(player, te, 166) {
    init {
        addSlotToContainer(SlotInventorySlot(te.slotRotor, 0, 80, 26))
    }
}